### DISCUSSION TITLE
redirectTo is disturbing the login with google feature

### DISCUSSION THREAD

**Discussion by hlassiege** [supabase/supabase#25952] [Questions] [answered]
Hi,

I'm using a google authentication like this : 

```
const signInWithOAuth = async () => {
    const { error } = await supabase.auth.signInWithOAuth({
        provider: 'google',
    })
    if (error) console.log(error)
}
```

And it works fine. I'm redirect to the home page

When I try to use the redirectTo option like this : 

```
const signInWithOAuth = async () => {
    const { error } = await supabase.auth.signInWithOAuth({
        provider: 'google',
        options: {
             redirectTo: 'http://localhost:3000/campaigns',
        }
    })
    if (error) console.log(error)
}
```

the user is not connected at all after the google auth workflow. I am then redirected from /campaigns to /login

It seems correctly configured on supabase also : 

![image](https://github.com/supabase/supabase/assets/666123/31dde07d-c669-46bd-ad37-9cc0794d12ce)

Any thoughts ?

**Comment by silentworks** [comment]
Are you doing this in a SSR framework like NextJS? if yes then the setup is a bit different and there are instructions in the guide here for that  https://supabase.com/docs/guides/auth/social-login/auth-google?queryGroups=environment&environment=server&queryGroups=framework&framework=nextjs

**Reply by hlassiege** [reply]
Hi,

It's NuxtJs, indeed it's an SSR framework. But the login flow is initiated on the client side  (https://www.rssfeedpulse.com/login) and the callback url is not an handler on the server side. 
It works without the redirectTo option. I have the feeling it should work with the redirect to option either, no ?

**Reply by hlassiege** [reply, marked as answer]
I tried to deep dive a little bit in the documentation you gave me and indeed, I changed : 

    options.queryParams = {
        access_type: 'offline',
        prompt: 'consent',
    }
    
  and it works !
  
  I don't understand why it worked before but, anyway. Thanks a lot !
